includes/specials/SpecialUserlogout.php

```diff
@@ -59,49 +62,33 @@ function execute( $par ) {
 		 *
 		 * Once the old-style global wgUser object is fully deprecated, this line can be removed.
 		*/
-		$wgUser->logout();	 /* wikia change */
+		$wgUser->logout();     /* wikia change */
 
 		// Wikia change
 		// regenerate session ID on user logout to avoid race conditions with
 		// long running requests logging the user back in (@see PLATFORM-1028)
 		wfResetSessionID();
 
-		$out = $this->getOutput();
+		wfRunHooks( 'UserLogoutComplete', array( &$user, null, $oldName ) );
+
+		// redirection
 
-		$loginUrl = ( F::app()->wg->EnableNewAuth && F::app()->checkSkin( 'wikiamobile' ) )
-			? ( new UserLoginHelper() )->getNewAuthUrl()
-			: SpecialPage::getTitleFor( 'UserLogin' )->getLocalURL();
-		$loginLink = '<a href="' . $loginUrl . '">' . wfMessage( 'logouttext-link-text' )->escaped() . '</a>';
-		$out->addHTML( wfMessage( 'logouttext' )->rawParams( $loginLink )->parse() );
-
-		// Hook.
-		$injected_html = '';
-		wfRunHooks( 'UserLogoutComplete', array( &$user, &$injected_html, $oldName ) );
-		$out->addHTML( $injected_html );
-
-		$mReturnTo = $this->getRequest()->getVal( 'returnto' );
-		$mReturnToQuery = $this->getRequest()->getVal( 'returntoquery' );
-
-		$title = Title::newFromText($mReturnTo);
-		if ( !empty($title) ) {
-			$mResolvedReturnTo = strtolower( array_shift( SpecialPageFactory::resolveAlias( $title->getDBKey() ) ) );
-			if ( in_array( $mResolvedReturnTo,array('userlogout','signup','connect') ) ) {
-				$titleObj = Title::newMainPage();
-				$mReturnTo = $titleObj->getText( );
-				$mReturnToQuery = '';
+		$headers = $this->getRequest()->getAllHeaders();
```

Comment:
Also, lines 76-84 seem like a good candidate for a new method, e.g. `getRedirectUrl`